src/ReTestItems.jl

```diff
@@ -415,70 +436,59 @@ function start_and_manage_worker(
                 ts = testitem_result.testset
                 push!(testitem.testsets, ts)
                 push!(testitem.stats, testitem_result.stats)
-                print_errors_and_captured_logs(testitem, nretries + 1; logs)
+                print_errors_and_captured_logs(testitem, run_number; logs)
                 report_empty_testsets(testitem, ts)
                 # Run GC to free memory on the worker before next testitem.
                 @debugv 2 "Running GC on $worker"
                 remote_fetch(worker, :(GC.gc(true); GC.gc(false)))
                 # If the result isn't a pass, we throw to go to the outer try-catch
                 throw_if_failed(ts)
                 testitem = next_testitem(testitems, testitem.id[])
-                nretries = 0
+                run_number = 1
             finally
                 close(timer)
             end
         catch e
             @debugv 2 "Error" exception=e
-            if !(e isa WorkerTerminatedException || e isa TimeoutException || e isa TestSetFailure)
-                # we don't expect any other kind of error, so rethrow, which will propagate
-                # back up to the main coordinator task and throw to the user
-                rethrow()
-            end
-
-            if !(e isa TestSetFailure)
-                println(DEFAULT_STDOUT[])
-                # Explicitly show captured logs or say there weren't any in case we're about
-                # to terminte the worker
-                _print_captured_logs(DEFAULT_STDOUT[], testitem, nretries + 1)
-            end
-
+            # Handle the exception
             if e isa TimeoutException
                 @debugv 1 "Test item $(repr(testitem.name)) timed out. Terminating worker $worker"
+                # Explicitly show captured logs or say there weren't any before terminating worker
+                println(DEFAULT_STDOUT[])
+                _print_captured_logs(DEFAULT_STDOUT[], testitem, run_number)
                 terminate!(worker)
                 wait(worker)
+                @warn "$worker timed out evaluating test item $(repr(testitem.name)) afer $timeout seconds. \
+                    Recording test error."
+                record_timeout!(testitem, run_number, timeout)
+            elseif e isa WorkerTerminatedException
+                @warn "$worker died evaluating test item $(repr(testitem.name)). \
+                    Recording test error."
+                record_worker_terminated!(testitem, run_number)
+            elseif e isa TestSetFailure
+                # We already printed the error and recorded the testset.
```

Comment:
where do we print the error? It seems we previously had a block that was removed above that did `println(DEFAULT_STDOUT[])`. were we double printing?

Reply:
we had 
```
            if !(e isa TestSetFailure)
                println(DEFAULT_STDOUT[])
                # Explicitly show captured logs or say there weren't any in case we're about
                # to terminte the worker
                _print_captured_logs(DEFAULT_STDOUT[], testitem, nretries + 1)
            end
```
but that moved into the `if e isa TimeoutException` block (since it doesn't actually apply in the worker crash case i don't think? but maybe it should be in that branch too?)
```
            if e isa TimeoutException
                @debugv 1 "Test item $(repr(testitem.name)) timed out. Terminating worker $worker"
                # Explicitly show captured logs or say there weren't any before terminating worker
                println(DEFAULT_STDOUT[])
                _print_captured_logs(DEFAULT_STDOUT[], testitem, run_number)
                terminate!(worker)
                wait(worker)
                ...
```

For the `e isa TestSetFailure` case, we only go to this whole `catch` block when we call `throw_if_failed`, and we have already called `print_errors_and_captured_logs` before that
Does this addition to the comment help?
```suggestion
                # We already printed the error and recorded the testset before calling `throw_if_failed`.
```

Reply:
I think we still want to try and print in the worker crash case, right @Drvi? like, ideally we get the segfault backtrace printed

Reply:
ah yeah, makes sense -- i was misled by the `in case we're about to terminate the worker` comment, which i thought meant these logs wouldn't exist if the worker had already terminated, but that's not the case (idk what that comment was meaning to say)

Reply:
>in case we're about to terminate the worker

I don't remember, but I think that at some point we just wanted to be extra sure we didn't lose any logs from dead workers, so we explicitly print "No Captured Logs" to signal that the reporting process didn't fail.